packages/core/src/lib/filter/index.ts

```diff
@@ -340,6 +340,8 @@ class Subscription {
   }
 }
 
+const DEFAULT_NUM_PEERS = 3;
```

Comment:
I know it is not a part of the PR but how is it enforced? It seems we do have this as default https://github.com/waku-org/js-waku/blob/fb41f4c723238628f398dff4bc423edee8fd1c6e/packages/sdk/src/utils/discovery.ts#L7

Reply:
These variables are for two separate uses:
- `DEFAULT_NUM_PEERS`: the number of peers to use for the protocol request (redundancy to increase decentralization)
- `DEFAULT_NODE_REQUIREMENTS`: the default configuration of number of peers for each protocol that we want to _discovery_ using DNS discovery

Reply:
but then it seems not to be used and fall back to `DEFAULT_NODE_REQUIREMENTS`, am I missing something?

Reply:
I'm not sure what you mean: the above two constants are for two very different purposes as described.

`numPeersToUse` falls back to `DEFAULT_NUM_PEERS` if not provided part of the `options` argument